MegaQuests/src/main/java/com/elmer/megaquests/commands/ResetCooldownCommand.java

```diff
@@ -1,40 +1,49 @@
 package com.elmer.megaquests.commands;
 
-import com.elmer.megaquests.MegaQuests;
+import com.elmer.megaquests.managers.CooldownManager;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandExecutor;
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
+import java.util.Objects;
+
 public class ResetCooldownCommand implements CommandExecutor {
-    MegaQuests megaQuests;
-    public ResetCooldownCommand(MegaQuests megaQuests){
-        this.megaQuests = megaQuests;
+    private final CooldownManager cooldownManager;
+
+    public ResetCooldownCommand(CooldownManager cooldownManager) {
+        this.cooldownManager = cooldownManager;
     }
+
     @Override
-    public boolean onCommand(CommandSender commandSender, Command command, String s, String[] args) {
-
-        if (commandSender instanceof Player){
-            Player player = (Player) commandSender;
-            if (player.hasPermission("megaquests.resetcooldown")){
-                if (args.length == 1){
-                    for (Player target : Bukkit.getOnlinePlayers()){
-                        if (target.getName().equals(args[0])){
-                            megaQuests.getCooldownManager().resetCooldown(target);
-                            player.sendMessage(ChatColor.GREEN + "Successfully reset " + player.getName() + "'s cooldown on new quests!");
-                        }
-                    }
-                } else {
-                    player.sendMessage(ChatColor.RED + "Invalid Usage! Please use /resetcooldown <player>");
-                }
-            } else {
-                player.sendMessage(ChatColor.RED + "You do not have permission to use this command!");
-            }
+    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
+
+        if (!(sender instanceof Player player)) {
+            sender.sendMessage(ChatColor.RED + "Only players can use this command!");
+            return false;
+        }
+
+        if (!player.hasPermission("megaquests.resetcooldown")) {
+            player.sendMessage(ChatColor.RED + "You do not have permission to use this command!");
+            return false;
+        }
+
+        if (args.length != 1) {
+            player.sendMessage(ChatColor.RED + "Invalid Usage! Please use /resetcooldown <player>");
+            return false;
+        }
 
+        String targetPlayerName = args[0];
+        Player targetPlayer = Bukkit.getPlayer(targetPlayerName);
+        if (Objects.isNull(targetPlayer)) {
+            player.sendMessage(ChatColor.RED + "Invalid Player!");
+            return false;

```

Comment:
Too much nesting.
It's very hard to read and very easy to get confused.
It is best to put the nesting into separate checks. 
In this case, I stop the code execution (return false) when the boundary case works. 
If everything is ok, i.e. player writes correct command, then code which comes after all checks will work.